### PR TITLE
threadlimit

### PR DESCRIPTION
**Pull request by DavidJanz**
Current implementation thrashes the cpu ---> extremely slow running the sweep.

Suggested fix here, but untested --- I get other errors trying to run the default agents:
1) transformed.apply expecting a random key, recent change in JAX, I suspect you need an older version of JAX / need to update this code
2) setup.py doesn't actually work for me out the box:
pip install bsuite[baselines]
ERROR: Could not find a version that satisfies the requirement tensorflow==2.1; extra == "baselines" (from bsuite[baselines]) (from versions: 2.2.0rc1, 2.2.0rc2, 2.2.0rc3, 2.2.0rc4, 2.2.0, 2.3.0rc0, 2.3.0rc1, 2.3.0rc2, 2.3.0)
ERROR: No matching distribution found for tensorflow==2.1; extra == "baselines" (from bsuite[baselines])

I use this in my code interacting with bsuite envs, it tends to speed it up by 10-100x. I suspect something like this might help on internal run scripts too.

### CODE REVIEW

**Comment by DavidJanz** [comment]
@googlebot I signed it!

**Comment by iosband** [comment]
Hi David - looks like there is something wrong with this pull request... I'm having a look if this improves performance

**Comment by DavidJanz** [comment]
Fixed a typo that was making actual tests fail. It should now be runnable. PyTest is failing, but I don't think that's anything to do with me? I made another pull request here which was just changing a doc string, and that also has PyTests failing.

**Comment by aslanides** [comment]
Hey David! There were some breakages upstream of us (rlax/chex) which should be fixed soon. I nthe mean time I have made a fix on our end, so this should be OK.

File: bsuite/baselines/utils/pool.py [hunk 1]
```diff
@@ -22,10 +22,18 @@
 
 import termcolor
 import tqdm
+import threadpoolctl
 
 BsuiteId = str
 
 
+def thread_limit_run_fn(run_fn, limit=1):
+  def _run_fn(*args, **kwargs)
```
Comment (aslanides) [inline bsuite/baselines/utils/pool.py:31]: This is missing a colon after the function signature.

Comment (DavidJanz) [reply bsuite/baselines/utils/pool.py]: Thanks, fixed.

**Comment by iosband** [comment]
I had a look at this change locally and within Google and I don't think I saw a benefit... bit confused what's going on here

**Comment by DavidJanz** [comment]
How are you running it? I'd suggest running it on a local machine for the test --- I have no idea what borg might be doing behind the scenes.

I'll put together an example script that times this stuff using the external interface over the next couple of days and send it over.

**Pull request closed without merging**